### ISSUE TITLE
vault-ruby incompatible with Vault 0.6.1

### ISSUE THREAD

**Issue by SarahKowalik** [hashicorp/vault-ruby#88]
We've upgraded to vault 0.6.1, and discovered that vault-ruby doesn't work correctly with it any more.

I've opened up a PR to work with the API data structure changes at https://github.com/hashicorp/vault-ruby/pull/87, but there are another 62 failing tests (https://travis-ci.org/hashicorp/vault-ruby/jobs/154903894).

Do you have an ETA on when vault-ruby will be updated to work with the released Vault server?

Thanks!

**Comment by grosser** [comment]
I'm seeing an issue where the response is no longer parsed as json ... did you see that too ?

ping @sethvargo

**Comment by SarahKowalik** [comment]
Which method?

**Comment by grosser** [comment]
`Vault::Logical#list`

```
TypeError: no implicit conversion of Symbol into Integer
       	from gems/vault-0.5.0/lib/vault/api/logical.rb:29:in `[]'
```

**Comment by grosser** [comment]
getting content type text/plain ... monkeypatch:

```Ruby
# vault returns text/plain and not the expected application/json
# https://github.com/hashicorp/vault-ruby/issues/88
Vault::Client.prepend(Module.new do
  def success(response)
    response.content_type = 'application/json' if response.body && response.body.start_with?('{', '[')
    super
  end
end)
```

**Comment by jefferai** [comment]
Are you using the beta request forwarding support?

**Comment by sethvargo** [comment]
Hi @SarahKowalik,

I fixed up all the Vault 0.6.1 issues and updated the test matrix to be quite verbose to prevent regressions in the future. All the tests are green now.

Hey @grosser,

All the tests are green, and I'm unable to reproduce your issue. Could you please share some additional details and a larger error/stack trace? Are you fronting Vault servers with some kind of LB that could be stripping the content-type? Are you using the new request-forwarding feature that's in beta?

**Comment by grosser** [comment]
issue is https://github.com/hashicorp/vault/issues/1791

**Comment by grosser** [comment]
will be fixed in vault 0.6.2

**Comment by mfischer-zd** [comment]
@sethvargo Can we get a new release with these fixes?
